### PR TITLE
Avoid null refs in BlazorIgntior when Disposed

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/14257

* Minor cleanup
* Update instructions to say chrome needs to be installed for Selenium

### CODE REVIEW

**Review by javiercn — COMMENTED** [review]
Wouldn't it be better to throw ObjectDisposedException if used after disposed? Or is this for calls that might come from the server?

**Review by javiercn — APPROVED** [review]
Wouldn't it be better to throw ObjectDisposedException if used after disposed? Or is this for calls that might come from the server?

**Comment by pranavkm** [comment]
Not sure I follow. When would we throw the ObjectDisposedException?

**Comment by javiercn** [comment]
> When would we throw the ObjectDisposedException?

When checking for Disposed?

**Comment by javiercn** [comment]
> > When would we throw the ObjectDisposedException?
> 
> When checking for Disposed?

Discussed offline. Pranav is doing the right thing.

File: src/Shared/E2ETesting/BrowserFixture.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ private async Task<(IWebDriver browser, ILogs log)> CreateBrowserAsync(string co
             var instance = await SeleniumStandaloneServer.GetInstanceAsync(output);
 
             var attempt = 0;
-            var maxAttempts = 3;
+            const int maxAttempts = 3;
```
Comment (rynowak) [inline src/Shared/E2ETesting/BrowserFixture.cs]: are we doing *this* now?

File: src/Components/Ignitor/src/BlazorClient.cs [hunk 2]
```diff
@@ -56,7 +56,7 @@ public BlazorClient()
 
         private CancellableOperation<CapturedAttachComponentCall> NextAttachComponentReceived { get; set; }
 
-        private CancellableOperation<CapturedRenderBatch> NextBatchReceived { get; set; }
+        internal CancellableOperation<CapturedRenderBatch> NextBatchReceived { get; set; }
```
Comment (rynowak) [inline src/Components/Ignitor/src/BlazorClient.cs]: Why is this internal?

Comment (pranavkm) [inline src/Components/Ignitor/src/BlazorClient.cs]:
```suggestion
        private CancellableOperation<CapturedRenderBatch> NextBatchReceived { get; set; }
```

File: src/Components/Ignitor/src/BlazorClient.cs [hunk 3]
```diff
@@ -396,7 +396,7 @@ private void OnBeginInvokeJS(int asyncHandle, string identifier, string argsJson
             NextJSInteropReceived?.Completion?.TrySetResult(null);
         }
 
-        private void OnRenderBatch(int id, byte[] data)
+        protected virtual void OnRenderBatch(int id, byte[] data)
```
Comment (pranavkm) [inline src/Components/Ignitor/src/BlazorClient.cs]:
```suggestion
        private void OnRenderBatch(int id, byte[] data)
```

File: src/Shared/E2ETesting/BrowserFixture.cs [hunk 4]
```diff
@@ -132,18 +132,16 @@ private async Task<(IWebDriver browser, ILogs log)> CreateBrowserAsync(string co
...
+                    output.WriteLine($"Error initializing RemoteWebDriver: {ex.Message}");
                 }
+
                 attempt++;
+
             } while (attempt < maxAttempts);
 
-            // We will never get here. Keeping the compiler happy.
-            throw new InvalidOperationException("Couldn't create a Selenium remote driver client. The server is unresponsive");
+            throw new InvalidOperationException("Couldn't create a Selenium remote driver client. The server is irresponsive");
```
Comment (SteveSandersonMS) [inline src/Shared/E2ETesting/BrowserFixture.cs]:
```suggestion
            throw new InvalidOperationException("Couldn't create a Selenium remote driver client. The server is anti-responsive");
```

(Joking, but I don't actually understand this change)

Comment (javiercn) [reply src/Shared/E2ETesting/BrowserFixture.cs]: Completely orthogonal comment, I've seen selenium timeout in some tests, I'm wondering if we have instances where it becomes irresponsible. And if we should change the code that initializes selenium (in the future) to do this check on a per-test basis and switch the server (start a new one) if it detects the current one is not being responsive.

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=365210&view=ms.vss-test-web.build-test-results-tab&runId=11048896&resultId=127258&paneView=debug

**Pull request merged**